test/goHome.test.ts

```diff
@@ -0,0 +1,102 @@
+import { chromium, Page, Browser, BrowserContext, Cookie } from "playwright"
+import { createCookieIfDoesntExist } from "../src/common/util"
+import { hash } from "../src/node/util"
+
+async function setTimeoutPromise(milliseconds: number): Promise<void> {
+  return new Promise((resolve, _) => {
+    setTimeout(() => {
+      resolve()
+    }, milliseconds)
+  })
+}
+
+describe("login", () => {
+  let browser: Browser
+  let page: Page
+  let context: BrowserContext
+
+  beforeAll(async () => {
+    browser = await chromium.launch()
+    // Create a new context with the saved storage state
+    const storageState = JSON.parse(process.env.STORAGE || "")
+
+    const cookieToStore = {
+      sameSite: "Lax" as const,
+      name: "key",
+      value: hash(process.env.PASSWORD || ""),
+      domain: "localhost",
+      path: "/",
+      expires: -1,
+      httpOnly: false,
+      secure: false,
+    }
+
+    // For some odd reason, the login method used in globalSetup.ts doesn't always work
+    // I don't know if it's on playwright clearing our cookies by accident
+    // or if it's our cookies disappearing.
+    // This means we need an additional check to make sure we're logged in.
+    // We do this by manually adding the cookie to the browser environment
+    // if it's not there at the time the test starts
+    const cookies: Cookie[] = storageState.cookies || []
+    // If the cookie exists in cookies then
+    // this will return the cookies with no changes
+    // otherwise if it doesn't exist, it will create it
+    // hence the name maybeUpdatedCookies
+    const maybeUpdatedCookies = createCookieIfDoesntExist(cookies, cookieToStore)
```

Comment:
The playwright storage thing sometimes works and sometimes doesn't right? Or does it never work?

Wondering if we should add an optional test here just to keep an eye on how often it fails.

Reply:
Doesn't look like there are optional tests so maybe we can just log.

Reply:
yeah no optional tests :( should I make a note to add more login tests (testing different strategies) at a later time (different PR)?

Reply:
Yeah let's just smack down a TODO to figure this out.